README.txt
Major restructuring to support our UCDavis website project

HISTORY.txt
Version History
===============

3.1.3 (2012-11-14)
------------------

* Added a MANIFEST.in to exclude the examples folder from being packaged with
  the egg to work around the SandboxViolation from EasyInstall when installing
  FSD via buildout on Windows systems.
  [cah190]

* Work around membrane shortcoming which caused group property providers to
  be enumerated as users, raising an error and preventing further user
  enumeration.
  [Laurence Rowe]


3.1.2 (2012-04-20)
------------------

* Added a 'use site default' option to the available editors list.  Thanks to
  David Carter for reporting this issue and supplying a patch.
  [cah190]

* Require Products.membrane >= 2.1.4 to get the property sheet typing fix which
  affects the ext_editor field.
  [cah190]


3.1.1 (2012-03-28)
------------------

* Added a sample-content profile. [pumazi]

* fsd_atoz_view and department_textual_view should now work on Department objects
  living outside the FacultyStaffDirectory root on a site.  Warning: external
  Departments will only work reliably for situations where there is a single
  FacultyStaffDirectory object in a site.  If there are multiple
  FacultyStaffDirectory objects, then use of external Departments will have
  undefined behavior.
  [cah190]

* Fixes to author.cpt to support Plone 4.x.  Thanks to Paul Roeland for reporting
  this problem and supplying a patch.
  [cah190]

* Fixed fsd_atoz_view to work with last names having first characters outside the
  A to Z range.  We now allow for any valid unicode character.  Thanks to
  Toni Haka-Risku for reporting this problem and supplying a patch which served
  as a starting point for this fix.
  [cah190]


3.1 (2012-03-16)
----------------

* Retagging as 3.1.
  [cah190]


3.1b1 (2011-08-18)
------------------

* Rework templates to reflect the Plone 4.0 layout changes.
  [esteele]

* All FSD content type interfaces now subclass IFacultyStaffDirectoryContent. 
  Q: Why didn't they do so previously?
  A: I have no idea.
  [esteele]

* Drop support for Plone releases below 4.0.
  [esteele]
  
* Remove the reliance on the documentactions viewlet (not shown by default in 
  Plone 4+). vCard link now appears below the contact information in
  person_view.
  [esteele]
  
  
3.0.1
-----

* Add FacultyStaffDirectory to the content types controlled by the FSD Tool's
  activeMembraneStates field. Corrects an issue with customized viewlets 
  failing their permission checks.
  [esteele, pumazi]

* Updated Spanish translation for plone and FacultyStaffDirectory domain
  [macagua (Leonardo J. Caballero G.)]

* Moved the Python import of zope.i18nmessageid to __init__.py and import 
  the MessageFactory for FacultyStaffDirectory domain as FSDMessageFactory
  [macagua (Leonardo J. Caballero G.)]

* Added more support for i18n to "default" profile generic setup.
  [macagua (Leonardo J. Caballero G.)]

3.0
---

* Update person.pt in MobilePhoneExtender to be consistent with FSD's person.pt
  (to fix Plone 4 compatibility).
  [cah190]

* Added missing adapter registration to MobilePhoneExtender.
  [cah190]
  
* activeMembraneStates in the fsdtool now uses a MultiSelectionWidget and the
  workflow states vocabulary.
  [cah190]
  
* Updated course_view.pt for Plone 4.
  [cah190]

3.0b4
-----

* No longer support extenderInstallation install methods.  installExtender and
  installExtenderGloballyIfLocallyIsNotSupported will now throw exceptions.
  uninstallExtender will stay around for at least the 3.0 releases to permit
  easy cleanup of old extender installs.
  [cah190]

* Extender developers should use IBrowserLayerAwareExtender to register their
  extender against a browserlayer to make it possible to activate the extender
  per Plone site.  MobilePhoneExtender has been updated as an example.
  [cah190]

* Drop the icon for the vCard action.
  [esteele]
  
* Slightly rearrange package layout.
  [esteele]
  
* Stop creating a 'my folder' action. This went away in Plone 3.
  [esteele]
  
* Stop using deprecated actionicons_tool and put the icon directly on the action.
  [cah190 (Craig Haynal)]

* Register facultystaffdirectory_tool via GS.
  [cah190 (Craig Haynal)]

* Catalog indexes and metadata are now loaded via GS.
  [cah190 (Craig Haynal)]

* Collection/SmartFolder indexes are now loaded via GS.
  [cah190 (Craig Haynal)]

* Register configlet via GenericSetup.
  [esteele]
  
* Set up navtree properties via GenericSetup.
  [esteele]
  
* Use an import step and GS profile to set up versioning and diffing,
  respectively.
  [esteele]
  
* Reinstall/Refresh is not a Plone supported updating mechanism. Use
  portal_setup instead. Removing the code encouraging this sort of thing.
  [esteele]
  
* Move kupu resource types registration to a GS import step. 
  [esteele]
  
* Drop the 1.0-2.0 migration script. We'll no longer support a direct 1.0-3.0
  migration.
  [esteele]

* Moved css registry manipulation to GS profile, dropped css and js registry
  install code.
  [cah190 (Craig Haynal)]
  
* Install dependencies via GS rather than through Install.py
  [cah190 (Craig Haynal)]
    
* Factory tool registration is now via GS.
  [cah190 (Craig Haynal)]

3.0b3
-----
* Dropped the plone3 GS profile, paved over the types/\*.xml with versions
  from a working FSD Plone instance.
  [cah190 (Craig Haynal)]

* Don't call Products.Archetypes.Extensions.utils helpers, they seem to make
  Archetypes move itself in the portal_skins lists in a way that breaks
  TinyMCE.
  [cah190 (Craig Haynal)]

* Installer should not be installing ATReferenceBrowserWidget.
  [cah190 (Craig Haynal)]

* Register our skins directory the modern way (through ZCML and skins.xml)
  [cah190 (Craig Haynal)]

* Pin to archetypes.schemaextender < 2.1 because 2.1 and higher use plone.uuid
  which breaks Products.Relations and older versions of Archetypes.
  [cah190 (Craig Haynal)]

* Fixed schema conditions and fsdtool data to prevent authorization failures

Products/FacultyStaffDirectory/migrations.py
import logging

from Products.CMFCore.utils import getToolByName
from plone.app.workflow.remap import remap_workflow


def from_2_x_to_3_0(context):
    log = logging.getLogger('FacultyStaffDirectory')
    
    context.runImportStepFromProfile('profile-Products.FacultyStaffDirectory:default','workflow')
    try:
        remap_workflow(context,
                       ('FSDFacultyStaffDirectory',),
                       ('fsd_directory_workflow',),
                       {})
    except Exception, message:
        log.error(message)
        raise

    # log it
    log = logging.getLogger("FacultyStaffDirectory")
    log.info("Upgraded version 2 to version 3.0b1")

def from_3_0b1_to_3_0b3(context):
    log = logging.getLogger("FacultyStaffDirectory")
    membrane_upgrade = context.upgradeInfo('membrane')
    if membrane_upgrade['required'] == True and membrane_upgrade['available'] == True:
        log.info("Upgrading membrane")
        context.upgradeProduct('membrane')
        
    context.runImportStepFromProfile('profile-Products.FacultyStaffDirectory:default','FacultyStaffDirectory-reindexFSDObjects')
    log.info("Upgraded version 3.0b1 to version 3.0b3")
    
def from_3_0b3_to_3_0b4(context):
    log = logging.getLogger("FacultyStaffDirectory")
    registry = context.getImportStepRegistry()
    
    # drop legacy import steps, if they exist
    LEGACY_IMPORT_STEPS = [u'upgrade_2_to_3', u'hideMemberPrefs', u'installKupuResources', u'installVersionedTypes', u'installRelationsRules', u'unindexFSDTool']
    for step in LEGACY_IMPORT_STEPS:
        if step in registry.listSteps() and registry.getStepMetadata(step)['handler'] == ('Products.FacultyStaffDirectory.setuphandlers.%s' % step):
            registry.unregisterStep(step)        
            log.info("Unregistered import step: %s" % step)

    context.runImportStepFromProfile('profile-Products.FacultyStaffDirectory:default','typeinfo')
    
    log.info("Upgraded version 3.0b3 to version 3.0b4")

def from_3_0_to_3_0_1(context):
    log = logging.getLogger("FacultyStaffDirectory")

    tool = getToolByName(context, 'facultystaffdirectory_tool')
    states = list(tool.getActiveMembraneStates())
    states.append('published')
    tool.setActiveMembraneStates(tuple(states))
    
    log.info("Upgraded version 3.0 to version 3.0.1")


def from_3_0_1_to_3_1(context):
    log = logging.getLogger("FacultyStaffDirectory")

    context.runImportStepFromProfile('profile-Products.FacultyStaffDirectory:default','typeinfo')
    try:
        context.runImportStepFromProfile('profile-Products.FacultyStaffDirectory:default','repositorytool')
    except ValueError:
        context.runImportStepFromProfile('profile-Products.FacultyStaffDirectory:default','FacultyStaffDirectory-installVersionedTypes')
    
    # Remove the vCard action
    ttool = getToolByName(context, 'portal_types')
    personType = ttool['FSDPerson']
    actions = personType.listActions()
    try:
        vcardindex = actions.index(personType.getActionObject('document_actions/vcard'))
    except ValueError:
        pass
    else:
        personType.deleteActions([vcardindex])
    
    log.info("Upgraded version 3.0.1 to version 3.1")
    